cmd/opampsupervisor/e2e_test.go

```diff
@@ -699,6 +716,31 @@ func TestSupervisorOpAMPConnectionSettings(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond, "Collector did not connect to new OpAMP server")
 }
 
+func TestSupervisorCanStartWithoutServerThenConnectLater(t *testing.T) {
+	var connectedToServer atomic.Bool
+	connectedToServer.Store(false)
+	initialServer := newUnstartedOpAMPServer(
+		t,
+		defaultConnectingHandler,
+		server.ConnectionCallbacksStruct{
+			OnConnectedFunc: func(_ context.Context, _ types.Connection) {
+				connectedToServer.Store(true)
+			},
+		})
+	defer initialServer.shutdown()
+
+	s := newSupervisor(t, "accepts_conn", map[string]string{"url": initialServer.addr})
+	defer s.Shutdown()
+
+	time.Sleep(11 * time.Second) // We wait until the supervisor gives up on connecting
```

Comment:
Not the cleanest solution, but could we make the timeout a private field on the Supervisor and update it for this test? Adding 10 seconds to the execution times for the test will be a bit rough going forward.

Reply:
Sorry, missed this. I see the issue now, since `Supervisor` is in a separate package, we don't have access to private fields.

Could we then add this to the `OpAMPServer` config object as something like `initial_connection_timeout` with a default of 10 seconds? That way we can set it from the config file.

Reply:
I think the point at which the timer is triggered shouldn't be much of an issue, but if it is there should be at least be a way to delay this test to wait for it.

Reply:
when we add a configuration parameter, we're actually changing a user-facing feature of the supervisor, and we risk users depending on it. At the moment, this is a test-related capability and it might be better to keep it that way?

We could add a parameter to `NewSupervisor` that is similar to `initial_connection_timeout` instead. WDYT? A ClockImpl has more consequences, but it's a more general solution for other tests... 

My personal opinion is that:
- A parameter to `NewSupervisor` is preferrable from a config parameter
- A ClockImpl or an initial_connection_timeout are... about equally preferrable.

WDYT?

Reply:
>when we add a configuration parameter, we're actually changing a user-facing feature of the supervisor, and we risk users depending on it. At the moment, this is a test-related capability and it might be better to keep it that way?

I agree we shouldn't expose test-related configuration options to users, thanks for calling that out. My intention was actually for this to be a real option available to users. I think it makes sense that someone may want to tune the amount of time the Supervisor waits before it starts a Collector when it's unable to connect to the OpAMP server; you may want to fail fast and get _something_ going, or may want to wait longer because network conditions can be unreliable. Since we're adding the ability for the Collector to be started if the server can't be reached, I think this PR would also be a decent place to add that option. Do you think that's reasonable?